src/cephadm/cephadm

```diff
@@ -4415,82 +4422,162 @@ def is_ipv6(address):
+        port_str = hasport[0]
+        addr = re.sub(r'^v\d+:', '', addr)  # strip off v1: or v2: prefix
+        base_ip = addr[0:-(len(port_str)) - 1]
+        addrv_args.append(EndPoint(base_ip, int(port_str)))
+
+    return addrv_args
+
+
+def parse_mon_ip(mon_ip: str) -> List[EndPoint]:
+    """Parse mon-ip param into a list of mon end points."""
+    r = re.compile(r':(\d+)$')
+    addrv_args = []
+    hasport = r.findall(mon_ip)
+    if hasport:
+        port_str = hasport[0]
+        base_ip = mon_ip[0:-(len(port_str)) - 1]
+        addrv_args.append(EndPoint(base_ip, int(port_str)))
+    else:
+        # Use fixd ports for ceph monitor
+        addrv_args.append(EndPoint(mon_ip, 3300))
+        addrv_args.append(EndPoint(mon_ip, 6789))
+
+    return addrv_args
+
+
+def build_addrv_params(addrv: List[EndPoint]) -> str:
+    """Convert mon end points (ip:port) into the format: [v[1|2]:ip:port1]"""
+    addr_arg = ''
+    if len(addrv) == 1:
+        ip = addrv[0].ip
+        port = addrv[0].port
+        if port == 6789:
+            addr_arg = f'[v1:{ip}]'
+        elif port == 3300:
+            addr_arg = f'[v2:{ip}]'
+        else:
+            logger.warning(f'Using msgr2 protocol for unrecognized port {port}')
+            addr_arg = f'[v2:{ip}]'
+    elif len(addrv) == 2:
+        ip = addrv[0].ip
+        addr_arg = f'[v2:{ip}:3300,v1:{ip}:6789]'
+    else:
+        logger.warning(f'provided addrv {addrv} has a bad length')
+    return addr_arg
+
+
+def get_public_net_from_cfg(ctx: CephadmContext) -> Optional[str]:
+    """Get mon public network from configuration file."""
+    cp = read_config(ctx.config)
+    if not cp.has_option('global', 'public_network'):
+        return None
+
+    # Ensure all public CIDR networks are valid
+    public_network = cp.get('global', 'public_network')
+    rc, _, err_msg = check_subnet(public_network)
+    if rc:
+        raise Error(f'Invalid public_network {public_network} parameter: {err_msg}')
+
+    # Ensure all public CIDR networks are configured locally
+    configured_subnets = set([x.strip() for x in public_network.split(',')])
+    local_subnets = set([x[0] for x in list_networks(ctx).items()])
+    for net in configured_subnets:
+        if net not in local_subnets:
+            raise Error(f'The provided public CIDR network {net} is not configured locally.')
+
+    # Ensure public_network is compatible with the provided mon-ip (or mon-addrv)
+    if ctx.mon_ip:
+        if not ip_in_subnets(ctx.mon_ip, public_network):
+            raise Error(f'The provided --mon-ip {ctx.mon_ip} does not belong to any public_network(s) {public_network}')
+    elif ctx.mon_addrv:
+        addrv_args = parse_mon_addrv(ctx.mon_addrv)
+        for addrv in addrv_args:
+            if not ip_in_subnets(addrv.ip, public_network):
+                raise Error(f'The provided --mon-addrv {addrv[0]} ip does not belong to any public_network(s) {public_network}')
```

Comment:
```suggestion
                raise Error(f'The provided --mon-addrv {addrv.ip} ip does not belong to any public_network(s) {public_network}')
```
Since addrv should be one of your new EndPoint objects?

Reply:
Fixed